src/python/pants/scm/subsystems/changed.py

```diff
@@ -38,7 +52,36 @@ def register_options(cls, register):
              help='Calculate changes since this tree-ish/scm ref (defaults to current HEAD/tip).')
     register('--diffspec',
              help='Calculate changes contained within given scm spec (commit range/sha/ref/etc).')
-    register('--include-dependees', choices=['none', 'direct', 'transitive'], default='none',
+    register('--include-dependees', type=IncludeDependees, default=IncludeDependees.none,
              help='Include direct or transitive dependees of changed targets.')
     register('--fast', type=bool,
              help='Stop searching for owners once a source is mapped to at least one owning target.')
+
+
+class ScmWrapper(datatype([('scm', SubclassesOf(Scm))])): pass
+
+
+class ChangedFilesResult(datatype([('changed_files', TypedCollection(Exactly(text_type)))])): pass
+
+
+@rule(ChangedFilesResult, [Select(ScmWrapper), Select(ChangedRequest)])
+def get_changed(scm, changed_request):
```

Comment:
This will be cached in the daemon, whereas this implementation body relies on the rule never being cached.

If we want to express this in a v2 `@rule`, we'll need to work out how to denote that this rule isn't cacheable, which is a kind of scary piece of work...

Reply:
I was just thinking we could expose the kwarg `cacheable=True` from `_make_rule()`: https://github.com/pantsbuild/pants/blob/32b123954ebf41794d0c901b2d5e424a66f8ed82/src/python/pants/engine/rules.py#L232-L233

Reply:
Did the above, setting `cacheable=False` for this rule.

Reply:
The `cacheable` argument currently doesn't actually do the thing it's meant to, so this still won't get re-run. See https://github.com/pantsbuild/pants/issues/6598

Reply:
Commented!